### PR TITLE
Update course tutorials to Lightning 2.0

### PR DESCRIPTION
**Pull request by awaelchli**
## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## What does this PR do?

Updates all notebooks to support Lightning 2.0.
This PR depends on #222 

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: course_UvA-DL/01-introduction-to-pytorch/Introduction_to_PyTorch.py [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+#   jupytext:
+#     cell_metadata_filter: -all
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.14.5
+# ---
+
```
Comment (awaelchli) [inline course_UvA-DL/01-introduction-to-pytorch/Introduction_to_PyTorch.py:12]:
```suggestion
```

File: course_UvA-DL/03-initialization-and-optimization/Initialization_and_Optimization.py [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+#   jupytext:
+#     cell_metadata_filter: -all
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.14.5
+# ---
+
```
Comment (awaelchli) [inline course_UvA-DL/03-initialization-and-optimization/Initialization_and_Optimization.py:12]:
```suggestion
```

File: course_UvA-DL/04-inception-resnet-densenet/Inception_ResNet_DenseNet.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+#   jupytext:
+#     cell_metadata_filter: -all
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.14.5
+# ---
+
```
Comment (awaelchli) [inline course_UvA-DL/04-inception-resnet-densenet/Inception_ResNet_DenseNet.py:12]:
```suggestion
```

File: course_UvA-DL/08-deep-autoencoders/Deep_Autoencoders.py [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+#   jupytext:
+#     cell_metadata_filter: -all
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.14.5
+# ---
+
```
Comment (awaelchli) [inline course_UvA-DL/08-deep-autoencoders/Deep_Autoencoders.py:12]:
```suggestion
```

File: course_UvA-DL/05-transformers-and-MH-attention/Transformers_MHAttention.py [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+#   jupytext:
+#     cell_metadata_filter: -all
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.14.5
+# ---
+
```
Comment (awaelchli) [inline course_UvA-DL/05-transformers-and-MH-attention/Transformers_MHAttention.py:12]:
```suggestion
```

File: course_UvA-DL/04-inception-resnet-densenet/Inception_ResNet_DenseNet.py [hunk 6]
```diff
@@ -345,10 +346,10 @@ def train_model(model_name, save_name=None, **kwargs):
         save_name = model_name
 
     # Create a PyTorch Lightning trainer with the generation callback
-    trainer = pl.Trainer(
+    trainer = L.Trainer(
         default_root_dir=os.path.join(CHECKPOINT_PATH, save_name),  # Where to save models
         # We run on a single GPU (if possible)
-        accelerator="gpu" if str(device).startswith("cuda") else "cpu",
+        accelerator=("cuda" if str(device) == "cuda:0" else "cpu"),
```
Comment (awaelchli) [inline course_UvA-DL/04-inception-resnet-densenet/Inception_ResNet_DenseNet.py:352]:
```suggestion
        accelerator="auto",
```

File: course_UvA-DL/05-transformers-and-MH-attention/Transformers_MHAttention.py [hunk 7]
```diff
@@ -976,14 +976,13 @@ def train_reverse(**kwargs):
     # Create a PyTorch Lightning trainer with the generation callback
     root_dir = os.path.join(CHECKPOINT_PATH, "ReverseTask")
     os.makedirs(root_dir, exist_ok=True)
-    trainer = pl.Trainer(
+    trainer = L.Trainer(
         default_root_dir=root_dir,
         callbacks=[ModelCheckpoint(save_weights_only=True, mode="max", monitor="val_acc")],
-        accelerator="gpu" if str(device).startswith("cuda") else "cpu",
+        accelerator=("cuda" if str(device).startswith("cuda") else "cpu"),
```
Comment (awaelchli) [inline course_UvA-DL/05-transformers-and-MH-attention/Transformers_MHAttention.py:982]:
```suggestion
        accelerator="auto",
```

File: course_UvA-DL/05-transformers-and-MH-attention/Transformers_MHAttention.py [hunk 8]
```diff
@@ -1437,14 +1436,13 @@ def train_anomaly(**kwargs):
     # Create a PyTorch Lightning trainer with the generation callback
     root_dir = os.path.join(CHECKPOINT_PATH, "SetAnomalyTask")
     os.makedirs(root_dir, exist_ok=True)
-    trainer = pl.Trainer(
+    trainer = L.Trainer(
         default_root_dir=root_dir,
         callbacks=[ModelCheckpoint(save_weights_only=True, mode="max", monitor="val_acc")],
-        accelerator="gpu" if str(device).startswith("cuda") else "cpu",
+        accelerator=("cuda" if str(device).startswith("cuda") else "cpu"),
```
Comment (awaelchli) [inline course_UvA-DL/05-transformers-and-MH-attention/Transformers_MHAttention.py:1442]:
```suggestion
        accelerator="auto",
```

File: course_UvA-DL/07-deep-energy-based-generative-models/Deep_Energy_Models.py [hunk 9]
```diff
@@ -638,9 +638,9 @@ def on_epoch_end(self, trainer, pl_module):
 # %%
 def train_model(**kwargs):
     # Create a PyTorch Lightning trainer with the generation callback
-    trainer = pl.Trainer(
+    trainer = L.Trainer(
         default_root_dir=os.path.join(CHECKPOINT_PATH, "MNIST"),
-        accelerator="gpu" if str(device).startswith("cuda") else "cpu",
+        accelerator=("cuda" if str(device).startswith("cuda") else "cpu"),
```
Comment (awaelchli) [inline course_UvA-DL/07-deep-energy-based-generative-models/Deep_Energy_Models.py:643]:
```suggestion
        accelerator="auto",
```

File: course_UvA-DL/08-deep-autoencoders/Deep_Autoencoders.py [hunk 10]
```diff
@@ -383,9 +383,9 @@ def on_train_epoch_end(self, trainer, pl_module):
 # %%
 def train_cifar(latent_dim):
     # Create a PyTorch Lightning trainer with the generation callback
-    trainer = pl.Trainer(
+    trainer = L.Trainer(
         default_root_dir=os.path.join(CHECKPOINT_PATH, "cifar10_%i" % latent_dim),
-        accelerator="gpu" if str(device).startswith("cuda") else "cpu",
+        accelerator=("cuda" if str(device).startswith("cuda") else "cpu"),
```
Comment (awaelchli) [inline course_UvA-DL/08-deep-autoencoders/Deep_Autoencoders.py:388]:
```suggestion
        accelerator="auto",
```

**Review by Borda — APPROVED** [review]

File: course_UvA-DL/01-introduction-to-pytorch/Introduction_to_PyTorch.py [hunk 11]
```diff
@@ -185,7 +185,7 @@
 print("X2 (after)", x2)
 
 # %% [markdown]
-# In-place operations are usually marked with a underscore postfix (e.g. "add_" instead of "add").
+# In-place operations are usually marked with a underscore postfix (e.g. "torch.add_" instead of "torch.add").
```
Comment (Borda) [inline course_UvA-DL/01-introduction-to-pytorch/Introduction_to_PyTorch.py]: you may need to make it as ``

**Pull request merged**